components/api_request_helper/api_request_helper.cc

```diff
@@ -94,6 +116,25 @@ bool APIRequestResult::IsResponseCodeValid() const {
   return response_code_ >= 100 && response_code_ <= 599;
 }
 
+base::Value APIRequestResult::TakeBody() {
+#if DCHECK_IS_ON()
+  DCHECK(!body_consumed_);
+  body_consumed_ = true;
+#endif
+  return std::move(value_body_);
+}
+
+std::string APIRequestResult::SerializeBodyToString() const {
+  if (value_body_.is_none()) {
+    return std::string();
+  }
+  std::string safe_json;
+  if (!base::JSONWriter::Write(value_body_, &safe_json)) {
+    VLOG(1) << "Response validation error: Encoding error";
```

Comment:
Is this really can happen?

Reply:
Not 100% sure, but in theory it's possible. `Write` returns `false` if `max_depth` is reacted. If we use different parser to make base::Value (e.g. for Android) looks possible.